### PR TITLE
Integration resource: add types and grpc methods

### PR DESCRIPTION
**Pull request by marcoandredinis**
This commit adds the new types for Integration resource. It also adds the gRPC methods that will be used later on for:
- Integrations CRUD management
- Integrations resource caching'

gRPC methods are not implemented yet.
Implementation will come in a later PR

RFD: https://github.com/gravitational/teleport/pull/22556

### CODE REVIEW

**Review by rosstimothy — COMMENTED** [review]
This would be a good opportunity to define a separate service instead of adding on to the AuthService. Look at https://github.com/gravitational/teleport/blob/master/api/proto/teleport/trust/v1/trust_service.proto and it's corresponding implementation in `lib/auth/trust` as an example.

File: api/proto/teleport/legacy/types/types.proto [hunk 1]
```diff
@@ -5265,6 +5265,34 @@ message OktaAssignmentActionTargetV1 {
...
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "",
+    (gogoproto.embed) = true
+  ];
+
+  // Spec is an Integration specification.
+  IntegrationSpecV1 Spec = 2 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "spec"
+  ];
```
[the remark this answers is not in the document]
Comment (marcoandredinis) [reply api/proto/teleport/legacy/types/types.proto:5286]: That's a good point :+1: 

I was trying to keep things as simple as possible.

I don't think we'll have another Integration subkind any time soon. (that would be GCP/Azure or another AWS connection method)
I expect the development efforts to target the increase of the integration maturity (more actions) and not their diversity. 

Saying that, I'm ok with changing this to a `one of` if you feel strongly about it.

File: api/proto/teleport/legacy/types/types.proto [hunk 2]
```diff
@@ -5263,6 +5263,42 @@ message OktaAssignmentActionTargetV1 {
...
+
+  // Header is the resource header.
+  ResourceHeader header = 1 [
+    (gogoproto.nullable) = false,
+    (gogoproto.jsontag) = "",
+    (gogoproto.embed) = true
+  ];
+
+  // Spec is an Integration specification.
+  IntegrationSpecV1 Spec = 2 [
```
Comment (kimlisa) [inline api/proto/teleport/legacy/types/types.proto:5281]: i think generally we prefer lowercase snaking? (but in this file i see a mixture of both)

File: api/proto/teleport/integration/v1/integration_service.proto [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  string next_key = 2;
+}
+
+// ListIntegrationsResponse a paginated list of Integrations.
+message ListIntegrationsResponse {
+  // Integrations is a list of Integrations.
+  repeated types.IntegrationV1 integrations = 1;
+  // NextKey is the key for the next page of Integrations.
+  string next_key = 2;
+  // TotalCount is the total number of resources available after filter, if any.
```
Comment (kimlisa) [inline api/proto/teleport/integration/v1/integration_service.proto:59]: i saw someone got confused by this explanation (they thought it meant the total in the page that we just retrieved `len(repeated types.IntegrationV1)`), i'm not sure if the below is any better.

```suggestion
  // TotalCount is the total number of integrations in all pages after filter, if any.
```

**Review by kimlisa — APPROVED** [review]
LGTM

File: api/proto/teleport/legacy/types/types.proto (hunk 2 above)
Comment (marcoandredinis) [reply api/proto/teleport/legacy/types/types.proto:5281]: I'll convert to `Header` and `Spec`
Seems the most common pattern for the top level fields of a resource.

**Review by rosstimothy — APPROVED** [review]

**Pull request merged**